addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-config-vars-service.js

```diff
@@ -338,23 +337,23 @@ class EnvironmentConfigVarsService extends Service {
     return _.isUndefined(doc) ? {} : doc.toPolicyDoc();
   }
 
-  async getEnvEgressStorePolicy(requestContext, { environment, egressStore, memberAccountId }) {
+  async getEnvEgressStorePolicy(requestContext, { environmentId }) {
+    const curUser = _.get(requestContext, 'principalIdentifier.uid');
+    const dataEgressService = await this.service('dataEgressService');
+    const egressStoreInfo = await dataEgressService.getEgressStoreInfo(environmentId);
+    const isEgressStoreOwner = egressStoreInfo.createdBy === curUser;
+    if (!isAdmin(requestContext) && !isEgressStoreOwner) {
+      throw this.boom.forbidden(
+        `You are not authorized to access the attached egress store. Please contact your administrator.`,
+        true,
+      );
+    }
+
+    // Create a policy with just the AssumeRole statement for the new role created in the main account
     const policyDoc = new StudyPolicy();
-    const pluginRegistryService = await this.service('pluginRegistryService');
-    const payload = {
-      requestContext,
-      container: this.container,
-      environmentScEntity: environment,
-      egressStore,
-      policyDoc,
-      memberAccountId,
-    };
-    const result = await pluginRegistryService.visitPlugins('study-access-strategy', 'provideEnvEgressStorePolicy', {
-      payload,
-    });
+    policyDoc.roleArns = [egressStoreInfo.roleArn];
```

Comment:
Had to attach the roleArn here since the policy document would've been empty otherwise.